### ISSUE TITLE
Parser produces FunctionNodes with FunctionNodes as function

### ISSUE THREAD

**Issue by FSMaxB** [josdejong/mathjs#923]
When parsing `a(b)(c)` then what I would expect to happen is an implicit multiplication between `(c)` and a function `a` with argument `b`.

What actually happens is really strange:
```json
{
	"fn": {
		"fn": {
			"name": "a"
		},
		"args": [
			{
				"name": "b"
			}
		]
	},
	"args": [
		{
			"name": "c"
		}
	],
	"comment": ""
}
```

There is a `FunctionNode` in the `fn` member of a `FunctionNode`.

**Comment by josdejong** [comment]
hm, that's what I would expect too, an implicit multiplication. I will look into it towards the end of the week.

**Comment by josdejong** [comment]
This issue is addressed in #1035, will be fixed/changed in v4.
